lib/simp/cli/commands/environment/new.rb

```diff
@@ -0,0 +1,163 @@
+
+        By default, this command will:
+
+          * create a new environment (–-skeleton)
+          * raise an error if an environment directory already exists
+
+        It can create a complete SIMP omni-environment with --puppet-env
+
+        Examples:
+
+             # Create a skeleton new development environment
+             simp env new development
+
+             # Link staging's Secondary and Writable env dirs to production
+             simp env new staging --link production
+
+             # Create a separate copy of production (will diverge over time)
+             simp env new newprod --copy production
+
+             # Create new omni environment
+             simp env new local_prod --puppetfile
+
+        Options:
+
+      HELP_MSG
+
+      opts.on('--skeleton',
+              '(default) Generate environments from skeleton templates.',
+              'Implies --puppetfile') do
+                TYPES.each do |type|
+                  options[:types][type][:strategy]    = :skeleton
+                end
+                options[:types][:puppet][:puppetfile_generate] = true
+              end
+
+      opts.on('--copy ENVIRONMENT', Simp::Cli::Utils::REGEXP_PUPPET_ENV_NAME,
+              'Copy assets from ENVIRONMENT') do |src_env|
+                TYPES.each do |type|
+                  options[:types][type][:strategy] = :copy
+                  options[:types][type][:src_env]  = src_env
+                end
+              end
+
+      opts.on('--link ENVIRONMENT', Simp::Cli::Utils::REGEXP_PUPPET_ENV_NAME,
+              'Symlink Secondary and Writeable environment directories',
+              'to ENVIRONMENT.  If --puppet-env is set, the Puppet',
+              'environment will --copy.') do |src_env|
+                TYPES.each do |type|
+                  options[:types][type][:strategy] = :link
+                  options[:types][type][:src_env]  = src_env
+                end
+                options[:types][:puppet][:strategy] = :copy
+              end
+
+      opts.on('--[no-]puppetfile',
+              'Generate Puppetfiles in Puppet env directory',
+              '  * `Puppetfile` will only be created if missing',
+              '  * `Puppetfile.simp` will be generated from RPM/',
+              '  * implies `--puppet-env`') do |v|
+        options[:types][:puppet][:enabled] = true if (options[:types][:puppet][:puppetfile_generate] = v)
+      end
+
+      opts.on('--[no-]puppetfile-install',
+              'Automatically deploys Puppetfile in Puppet environment',
+              'directory after creating it',
+              '  * implies `--puppet-env`',
+              '  * Does NOT imply `--puppetfile`') do |v|
+        options[:types][:puppet][:enabled] = true if (options[:types][:puppet][:puppetfile_install] = v)
+      end
+
+      opts.on('--[no-]puppet-env',
+              'Includes Puppet environment when `--puppet-env`',
+              '(default: --no-puppet-env)') { |v| options[:types][:puppet][:enabled] = v }
```

Comment:
Default of --no-puppet-env?  When I executed the first example (no command line options), the puppet-env skeleton was created, because --skeleton implies --puppetfile which implies --puppet-env.   I guess this is saying --no-puppet-env is the default when --copy or --link is called?